src/industry_gui.cpp

```diff
@@ -158,7 +158,7 @@ static inline void GetAllCargoSuffixes(CargoSuffixInOut use_input, CargoSuffixTy
 	if (indspec->behaviour & INDUSTRYBEH_CARGOTYPES_UNLIMITED) {
 		/* Reworked behaviour with new many-in-many-out scheme */
 		for (uint j = 0; j < lengthof(suffixes); j++) {
-			if (IsValidCargoID(cargoes[j])) {
+			if (j < lengthof(cargoes) && IsValidCargoID(cargoes[j])) {
```

Comment:
Does `lengthof` work with `std::array`? Why not `j < cargoes.size()`? Similarly for the line above it.
I think cargoes and suffixes got changed to `std::array`, so that should be fine right?

Reply:
This function is now only used against the industry spec which is still using C-style arrays. So it's out of scope here (originally I wrangled the industry cargoes into it), and not even necessary.